docs/tool/js/views/map-view.js

```diff
@@ -869,6 +871,59 @@
                 .remove();
 
         },
+        exportCsv: function() {
+          // if (data){
+            d3.select('#csvExport')
+              .on('click', function(d) {
```

Comment:
It would be best to move this anonymous function out so that it is an independent function that can be called from anywhere else in the code. This better matches the pattern we're using for the rest of the code, and allows us to more easily expand the functionality in the future. For instance, we could put multiple buttons in different parts of the website and they could all perform this same functionality. While were at it, better to adjust how the function is set up too. 

- [ ] remove the d3.select on click part of the code, so that exportCsv is just a function that does the exporting whenever it is called
- [ ] add another function called 'addExportButton' and use it to add the on click event to the button. 
- [ ] in init section, change this line: ` ['filteredData',mapView.exportCsv],` to instead by this line: ` ['filteredViewLoaded',mapView.addExportButton],`. This means the on click event will only be attached to the button one time.

Reply:
Done